code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -407,7 +410,69 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 
 	return ""
 
-/datum/gas_mixture/react(datum/holder)
+/datum/gas_mixture/proc/new_react(datum/holder)
+	. = NO_REACTION
+	var/list/cached_gases = gases
+	if(!length(cached_gases))
+		return
+	var/topprio // priority of the first element in the list
+	var/list/reactions = list()
+	for(var/I in cached_gases)
+		for(var/R in GLOB.reactionsdict[I])
+			var/datum/gas_reaction/reaction = R
+			if (reaction.priority > topprio)
+				topprio = reaction.priority
+				reactions.Insert(1, reaction)
+			else
+				reactions += reaction
+	if(!length(reactions))
+		return
+	reaction_results = new
+	var/temp = temperature
+	var/ener = THERMAL_ENERGY(src)
+
+	reaction_loop:
+		for(var/r in reactions)
+			var/datum/gas_reaction/reaction = r
+
+			var/list/min_reqs = reaction.min_requirements
+			if((min_reqs["TEMP"] && temp < min_reqs["TEMP"]) \
+			|| (min_reqs["ENER"] && ener < min_reqs["ENER"]))
+				continue
+
+			for(var/id in min_reqs)
+				if (id == "TEMP" || id == "ENER")
+					continue
+				if(!cached_gases[id] || cached_gases[id][MOLES] < min_reqs[id])
+					continue reaction_loop
+			//at this point, all minimum requirements for the reaction are satisfied.
+
+			/*	currently no reactions have maximum requirements, so we can leave the checks commented out for a slight performance boost
+				PLEASE DO NOT REMOVE THIS CODE. the commenting is here only for a performance increase.

```

Comment:
`#ifdef`?

Reply:
not even, it's just literally unused code rn, but may be in the future

Reply:
it's unused code dunc wants to keep around